### ISSUE TITLE
Apply multiple docker image tags when publishing images

### ISSUE THREAD

**Issue by skuzzle** [spring-projects/spring-boot#20560]
As of  #19698 spring boot has native support for building docker images which is a great approach to further simplify the build process.

When publishing docker images a common approach is to provide multiple tags per image. Probably most common is to publish the actual version along with an additional `latest` tag. I'd love to see Spring Boot support multiple tags per image (that would also simplify migrating from the also great `jib` plugin which already supports multiple tags)

**Comment by scottfrederick** [comment]
This is a good idea. As you can see in #19698, we haven't yet gotten to publishing generated docker images to a repository via the Spring Boot build tool plugins. This is a lower-priority feature as there are plenty of ways to do this with other tools like `docker` CLI, and publishing images from the Spring Boot build tool plugins might not make it into 2.3.0.RELEASE. I'll tag this as a useful features of publishing if/when we do get to it.

**Comment by philwebb** [comment]
Blocked on #21001

**Comment by thadc23** [comment]
Now that the spring-boot-maven-plugin supports pushing native images to remote repos, should this be revisited? It would be very beneficial to be able to tag the build with the commit SHA and `latest` at the same time.

**Comment by scottfrederick** [comment]
Closing in favor of #27613
